packages/camera/ios/Classes/CameraPlugin.m

```diff
@@ -244,6 +233,59 @@ - (void)captureToFile:(NSString *)path result:(FlutterResult)result {
                                                            cameraPosition:_captureDevice.position]];
 }
 
+- (void)setCaptureSessionPreset:(NSString *)resolutionPreset {
+  int presetIndex;
+  if ([resolutionPreset isEqualToString:@"high"]) {
+    presetIndex = 0;
+  } else if ([resolutionPreset isEqualToString:@"medium"]) {
+    presetIndex = 2;
+  } else {
+    NSAssert([resolutionPreset isEqualToString:@"low"], @"Unknown resolution preset %@",
+             resolutionPreset);
+    presetIndex = 3;
+  }
+
+  switch (presetIndex) {
+    case 0:
```

Comment:
Any reason not to just case off of `resolutionPreset` directly here?

Reply:
Switch statements don't work with strings in Objective-c. They only work with primitive types.